// remark-fff@1.0.0-alpha.1/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  default: () => src_default,
  fs: () => fs,
  git: () => git,
  path: () => path,
  remarkFFF: () => remarkFFF
});
module.exports = __toCommonJS(src_exports);
var import_fff_flavored_frontmatter = require("fff-flavored-frontmatter");

// src/autofill.ts
var autofill_exports = {};
__export(autofill_exports, {
  fs: () => fs,
  git: () => git,
  path: () => path
});
var import_node_child_process = require("child_process");
var import_node_fs = require("fs");
var import_node_os = require("os");
var path = {
  urara: (path2) => path2.replace("/src/routes/", "/urara/")
};
var fs = (path2) => {
  const { ctime, mtime } = (0, import_node_fs.statSync)(path2);
  return {
    created: ({ created }) => created ?? ctime,
    updated: ({ updated }) => updated ?? mtime
  };
};
var git = (path2) => ({
  created: ({ created }) => created ?? (0, import_node_child_process.execFileSync)(
    "git",
    ["log", "--diff-filter=A", "--follow", "--format=%ai", "--", path2],
    { encoding: "utf8" }
  ).trim().split(import_node_os.EOL).pop(),
  updated: ({ updated }) => updated ?? (0, import_node_child_process.execFileSync)("git", ["log", "-1", "--pretty=format:%ai", "--", path2], {
    encoding: "utf8"
  }).trim()
});

// src/index.ts
var remarkFFF = (options = {
  presets: [],
  target: "mdsvex"
}) => (_tree, file) => {
  const post = file;
  const path2 = post.filename ?? post.path;
  const fm = (0, import_fff_flavored_frontmatter.transform)(
    {
      /* eslint-disable unicorn/no-nested-ternary */
      ...options.target === "mdsvex" ? post.data.fm : options.target === "astro" ? post.data.astro.frontmatter : options.target === "nuxt" ? post.data : post.data
      /* eslint-enable unicorn/no-nested-ternary */
    },
    [
      ...options.presets,
      ...options.autofill?.provider ? [
        autofill_exports[options.autofill.provider](
          options.autofill.path ? typeof options.autofill.path === "function" ? options.autofill.path(path2) : path[options.autofill.path](path2) : post.filename
        )
      ] : [],
      ...options.strict ? [(0, import_fff_flavored_frontmatter.strict)(options.strict)] : []
    ]
  );
  switch (options.target) {
    case "mdsvex": {
      file.data.fm = fm;
      break;
    }
    case "astro": {
      file.data.astro = {
        ...file.data.astro,
        frontmatter: fm
      };
      break;
    }
    case "nuxt": {
      file.data = fm;
      break;
    }
    default: {
      file.data = fm;
    }
  }
};
var src_default = remarkFFF;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  fs,
  git,
  path,
  remarkFFF
});
